_posts/pl/2024-07-12-tworzenie-wydajnych-obrazów-dockerowych-w-springu.md

```diff
@@ -0,0 +1,329 @@
+<​missing>      8 days ago           ENV JAVA_VERSION=jdk-17.0.11+9                  0B
+<​missing>      8 days ago           RUN /bin/sh -c set -eux;     apk add --no-ca…   17.3MB
+<​missing>      8 days ago           ENV LANG=en_US.UTF-8 LANGUAGE=en_US:en LC_AL…   0B
+<​missing>      8 days ago           ENV PATH=/opt/java/openjdk/bin:/usr/local/sb…   0B
+<​missing>      8 days ago           ENV JAVA_HOME=/opt/java/openjdk                 0B
+<​missing>      3 months ago         /bin/sh -c #(nop)  CMD ["/bin/sh"]              0B
+<​missing>      3 months ago         /bin/sh -c #(nop) ADD file:37a76ec18f9887751…   7.37MB
+```
+
+4. Wykonujemy zmianę kodu źródłowego aplikacji i ponownie budujemy obraz z nowym tagiem:
+```shell
+docker build . --tag service2
+```
+5. Wykonujemy polecenie listujące warstwy:
+```
+docker history service2
+```
+6. Rezultat polecenia:
+```
+IMAGE          CREATED          CREATED BY                                      SIZE
+e027785c6f71   34 seconds ago   /bin/sh -c #(nop)  ENTRYPOINT ["java" "org.s…   0B
+<​missing>      34 seconds ago   /bin/sh -c #(nop) COPY dir:0c4cebea0bf1ba4e8…   3.7kB
+<​missing>      2 minutes ago    /bin/sh -c #(nop) COPY dir:f782fe956cf5892f5…   0B
+<​missing>      2 minutes ago    /bin/sh -c #(nop) COPY dir:3d769b9b5528fa54f…   387kB
+<​missing>      2 minutes ago    /bin/sh -c #(nop) COPY dir:24195f786b612de17…   19.5MB
+<​missing>      2 minutes ago    /bin/sh -c #(nop) WORKDIR /app                  0B
+<​missing>      8 days ago       ENTRYPOINT ["/__cacert_entrypoint.sh"]          0B
+<​missing>      8 days ago       COPY entrypoint.sh /__cacert_entrypoint.sh #…   1.17kB
+<​missing>      8 days ago       RUN /bin/sh -c set -eux;     echo "Verifying…   0B
+<​missing>      8 days ago       RUN /bin/sh -c set -eux;     ARCH="$(apk --p…   140MB
+<​missing>      8 days ago       ENV JAVA_VERSION=jdk-17.0.11+9                  0B
+<​missing>      8 days ago       RUN /bin/sh -c set -eux;     apk add --no-ca…   17.3MB
+<​missing>      8 days ago       ENV LANG=en_US.UTF-8 LANGUAGE=en_US:en LC_AL…   0B
+<​missing>      8 days ago       ENV PATH=/opt/java/openjdk/bin:/usr/local/sb…   0B
+<​missing>      8 days ago       ENV JAVA_HOME=/opt/java/openjdk                 0B
+<​missing>      3 months ago     /bin/sh -c #(nop)  CMD ["/bin/sh"]              0B
+<​missing>      3 months ago     /bin/sh -c #(nop) ADD file:37a76ec18f9887751…   7.37MB
+```
+
+Na powyższym przykładzie widzimy, że zależności naszej aplikacji ważą około 20 MB. Wykonując drugi obraz, warstwa zależności została pobrana z cache, dzięki czemu zamiast wykorzystania 40 MB na dwa obraz wykorzystamy 20 MB.
+
+
+## Definiowanie własnych warstw zależności
+
+Jeżeli domyślna konfiguracja warstw jest dla nas niewystarczająca, jesteśmy w stanie określić własne warstwy zawierające konkretne zależności. Pozwoli to na wydzielenie zależności, co do których jesteśmy pewni, że są one podstawowe i na przykład wykorzystane w kilku projektach.
```

Comment:
"że są one podstawowe i na przykład wykorzystane w kilku projektach." - właściwie nigdzie nie piszemy o reużywalności tego rozwiązania pomiędzy projektami i może nie idźmy tą ścieżką (bo to raczej temat na kolejny artykuł). Czy lepszym argumentem nie byłoby "rzadko się zmieniają"